### PR TITLE
Update hillslope testing fsurdat to 5.3

### PR DESCRIPTION
**Pull request by samsrabin**
Resolves ESCOMP/CTSM#2748

- [x] `hillslope` test suite, Derecho: Completes successfully
- [x] `rimport`
- [x] `hillslope` test suite, Izumi: Completes successfully

### CODE REVIEW

**Comment by samsrabin** [comment]
@slevis-lmwg This is ready!

File: cime_config/testdefs/testmods_dirs/clm/Hillslope/user_nl_clm [hunk 1]
```diff
@@ -6,6 +6,6 @@ hillslope_transmissivity_method = 'LayerSum'
 hillslope_pft_distribution_method = 'PftLowlandUpland'
 hillslope_soil_profile_method = 'Uniform'
 
-fsurdat = '$DIN_LOC_ROOT/lnd/clm2/testdata/surfdata_10x15_hist_2000_78pfts_c240216.synthetic_hillslopes.nc'
+fsurdat = '$DIN_LOC_ROOT/lnd/clm2/testdata/surfdata_10x15_hist_2000_78pfts_c240905.synthetic_hillslopes3.nc'
```
Comment (slevis-lmwg) [inline cime_config/testdefs/testmods_dirs/clm/Hillslope/user_nl_clm]: Thank you @samsrabin!

A correction though:
@ekluzek and I decided to group all synthetic fsurdat/landuse files in `.../lnd/clm2/surfdata_esmf/ctsm5.3.0/synthetic`

Comment (slevis-lmwg) [reply cime_config/testdefs/testmods_dirs/clm/Hillslope/user_nl_clm]: ...as in https://github.com/ESCOMP/CTSM/issues/2515

**Review by slevis-lmwg — CHANGES_REQUESTED** [review]
I am requesting a change in file path.

File: cime_config/testdefs/testmods_dirs/clm/Hillslope/user_nl_clm (hunk 1 above)
Comment (slevis-lmwg) [reply cime_config/testdefs/testmods_dirs/clm/Hillslope/user_nl_clm]: @ekluzek please correct me if I misunderstood.

Comment (samsrabin) [reply cime_config/testdefs/testmods_dirs/clm/Hillslope/user_nl_clm]: Right, sorry!

**Comment by samsrabin** [comment]
Re-testing...

**Comment by samsrabin** [comment]
Done and passed!

**Review by slevis-lmwg — APPROVED** [review]
Approving and then merging. Thanks again, @samsrabin.

**Pull request merged**